File: MainActivity.java
package com.example.qiuduijifen;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void click(View btn){
        TextView scoreText = findViewById(R.id.scorel);
        int s1= Integer.parseInt(scoreText.getText().toString());
        if(btn.getId()==R.id.btn3){
            scoreText.setText(String.valueOf(i:s1+3));
        }else if(btn.getId()==R.id.btn2){
            scoreText.setText(String.valueOf(i:s1+2));
        }else if(btn.getId()==R.id.btn1){
            scoreText.setText(String.valueOf(i:s1+1));
        }

    }
    public void click(View btn){
        TextView scoreText = findViewById(R.id.scorelb);
        int s2= Integer.parseInt(scoreText.getText().toString());
        if(btn.getId()==R.id.btnb3){
            scoreText.setText(String.valueOf(i:s2+3));
        }else if(btn.getId()==R.id.btnb2){
            scoreText.setText(String.valueOf(i:s2+2));
        }else if(btn.getId()==R.id.btnb1){
            scoreText.setText(String.valueOf(i:s2+1));
        }

    }
    public void resetClick(View btn) {
        TextView scoreText = findViewById(R.id.scorel);
        scoreText.setText(String.valueOf(i:0));

    }
    public void resetClick(View btn) {
        TextView scoreText = findViewById(R.id.scorelb);
        scoreText.setText(String.valueOf(i:0));

    }
}

File: MainActivity.kt
package com.example.bmiceshi

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val editTextWeight = findViewById<EditText>(R.id.editTextWeight)
        val editTextHeight = findViewById<EditText>(R.id.editTextHeight)
        val buttonCalculate = findViewById<Button>(R.id.buttonCalculate)
        val textViewResult = findViewById<TextView>(R.id.textViewResult)
        buttonCalculate.setOnClickListener {
            val weightString = editTextWeight.text.toString()
            val heightString = editTextHeight.text.toString()
            if (!weightString.isEmpty() && !heightString.isEmpty()) {
                val weight = weightString.toDouble()
                val height = heightString.toDouble()
                val bmi = weight / (height * height)
                textViewResult.text = String.format("%.2f", bmi) + " BMI"
                if (bmi < 18.5) {
                    textViewResult.append("\n低体重多吃")
                } else if (bmi < 24.9) {
                    textViewResult.append("\n正常，保持")
                } else if (bmi < 29.9) {
                    textViewResult.append("\n高体重少吃")
                } else {
                    textViewResult.append("\n胖胖不好")
                }
            }
        }
    }
}

